pycaret/internal/pycaret_experiment/pycaret_experiment.py

```diff
@@ -447,287 +530,46 @@ def pull(self, pop=False) -> pd.DataFrame:  # added in pycaret==2.2.0
         Returns
         -------
         pandas.DataFrame
-            Equivalent to get_config('display_container')[-1]
 
         """
-        return self.display_container.pop(-1) if pop else self.display_container[-1]
+        return self._display_container.pop(-1) if pop else self._display_container[-1]
 
     @property
     def dataset(self):
         """Complete dataset without ignored columns."""
         return self.data[[c for c in self.data.columns if c not in self._fxs["Ignore"]]]
 
-    @property
-    def train(self):
-        """Training set."""
-        return self.dataset.loc[self.idx[0], :]
-
-    @property
-    def test(self):
-        """Test set."""
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            return self.dataset.loc[self.idx[1], :]
-        else:
-            # Return the y_test indices not X_test indices.
-            # X_test indices are expanded indices for handling FH with gaps.
-            # But if we return X_test indices, then we will get expanded test
-            # indices even for univariate time series without exogenous variables
-            # which would be confusing. Hence, we return y_test indices here and if
-            # we want to get X_test indices, then we use self.X_test directly.
-            # Refer:
-            # https://github.com/sktime/sktime/issues/2598#issuecomment-1203308542
-            # https://github.com/sktime/sktime/blob/4164639e1c521b112711c045d0f7e63013c1e4eb/sktime/forecasting/model_evaluation/_functions.py#L196
-            return self.dataset.loc[self.idx[1], :]
-
     @property
     def X(self):
         """Feature set."""
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            if self.target_param:
-                return self.dataset.drop(self.target_param, axis=1)
-            else:
-                return self.dataset  # For unsupervised: dataset == X
-        else:
-            X = self.dataset.drop(self.target_param, axis=1)
-            if X.empty and self.fe_exogenous is None:
-                return None
-            else:
-                # If X is not empty or empty but self.fe_exogenous is provided
-                # Return X instead of None, since the index can be used to
-                # generate features using self.fe_exogenous
-                return X
+        return self.dataset
```

Comment:
Would this return the y values as well?

Reply:
this is overriden in `SupervisedExperiment` to return only X values